utilities/pulumi_state_splitter/pulumi_state_splitter/state_file.py

```diff
@@ -0,0 +1,94 @@
+"""Manipulation of the Pulumi stack state file."""
+
+import pathlib
+from typing import Iterable, List
+
+import pulumi_state_splitter.fs
+import pulumi_state_splitter.model
+import pulumi_state_splitter.stored_state
+
+
+def sorted_resources(
+    resources: Iterable[pulumi_state_splitter.model.Resource],
+) -> List[pulumi_state_splitter.model.Resource]:
+    """Sorts resources to Pulumi's liking."""
+    # sort by URN first
+    urn2resource = dict(sorted((resource.urn, resource) for resource in resources))
+
+    output = {}
+
+    def dependencies_first(resource: pulumi_state_splitter.model.Resource):
+        dependencies = resource.dependencies.copy()
+        if resource.provider:
+            provider_urn = resource.provider.rsplit("::", 1)[0]
+            dependencies.append(provider_urn)
+        if resource.parent:
+            dependencies.append(resource.parent)
+        dependencies.sort()
+        for dependency in dependencies:
+            dependencies_first(urn2resource[dependency])
+        output.setdefault(resource.urn, resource)
+
+    for resource in urn2resource.values():
+        dependencies_first(resource)
+    return list(output.values())
+
+
+class StateFile(pulumi_state_splitter.stored_state.StoredState):
+    """Represents a Pulumi stack state file."""
+
+    @property
+    def path(self) -> pathlib.Path:
+        """Path of the state file."""
+        return (
+            self.backend_dir / ".pulumi" / "stacks" / str(self.stack_name)
+        ).with_suffix(".json")
+
+    @classmethod
+    def find(
+        cls, backend_dir: pathlib.Path
+    ) -> Iterable[pulumi_state_splitter.stored_state.StackName]:
+        """Finds all file states in the Pulumi backend directory."""
+        glob_states = cls._glob_all().path
+        for stack_path in backend_dir.glob(str(glob_states)):
+            yield pulumi_state_splitter.stored_state.StackName(
+                project=stack_path.parent.name,
+                stack=stack_path.with_suffix("").name,
+            )
+
+    def remove(self):
+        self.path.unlink()
+        for d in (
+            self.path.parent,
+            self.path.parent.parent,
+            self.path.parent.parent.parent,
+        ):
```

Comment:
Nope, the organization part is only supported for `StackName`, and since Pulumi won't accept any other organization than `organization` for `file://` backends, we do not put the organization name in the paths. It would have to always be `organization`.